node/session.go

```diff
@@ -140,10 +140,16 @@ func (s *Session) sendFrame(frame *sentFrame) {
 
 	select {
 	case s.send <- *frame:
+		if s.node != nil {
+			s.node.Metrics.Counter(metricsSentMsg).Inc()
```

Comment:
I don't think that's the right place to increment this counter: sending to the `s.send` channel is not equal to sending to the client; we should better add this after we call `Write` (in `SendMessages`) and when it doesn't return an error.

And `metricsFailedSent` should be handled in `SendMessages` as well.

Reply:
I've moved the code to SendMessages and also removed the nil check.